sway-core/src/type_system/info.rs

```diff
@@ -1751,6 +1751,68 @@ impl TypeInfo {
             }
         }
     }
+
+    // Return a string representing only the base type.
+    // This is used by the trait map to filter the entries into a HashMap with the return type string as key.
+    pub fn get_type_str_filter(&self, engines: &Engines) -> String {
```

Comment:
We need some unique recognizable identifier of the base type here, can it not just be an enum that simply strips the info we don't want?

Reply:
Still, we don't need to be doing these string conversion for every builtin type, and it would make us explicit what the selector is. I think it'd be cleaner to have a specific type to hold that information.

Reply:
For me the hard part is to understand what is a "base type". If `TypeInfoBaseType` were an enum with Eq, Hash etc... and some comments why its needed would really help. We can even come up wit ha better name for it.